src/test/java/test/de/smartics/maven/plugin/jboss/modules/parser/ModulesXmlParserMatchXmlV2FilterTest.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2013-2017 smartics, Kronseder & Reiner GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package test.de.smartics.maven.plugin.jboss.modules.parser;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+
+import java.util.List;
+
+import org.junit.Test;
+
+import de.smartics.maven.plugin.jboss.modules.descriptor.ArtifactClusion;
+import de.smartics.maven.plugin.jboss.modules.descriptor.ArtifactMatcher;
+import de.smartics.maven.plugin.jboss.modules.descriptor.Directives;
+import de.smartics.maven.plugin.jboss.modules.descriptor.ModuleDescriptor;
+
+/**
+ * Tests {@link de.smartics.maven.plugin.jboss.modules.parser.ModulesXmlParser}
+ * with <code>{@value #ID}</code>.
+ */
+public class ModulesXmlParserMatchXmlV2FilterTest extends AbstractModulesXmlParserTest
+{
+  // ********************************* Fields *********************************
+
+  // --- constants ------------------------------------------------------------
+
+  private static final String ID = "match-v2-filter.xml";
+
+  // --- members --------------------------------------------------------------
+
+  // ****************************** Constructors ******************************
+
+  public ModulesXmlParserMatchXmlV2FilterTest()
```

Comment:
Does the filter take effect exclusively on the V2 schema? Or can it also operate on the V1 schema defintions? 

If it can work with V1, then we should add unit and integration tests for it (together with a XSD schema update).

Reply:
> Does the filter take effect exclusively on the V2 schema?

Yes

> can it also operate on the V1 schema defintions?

No

It is not a good idea to change XML schemas once they are published and potentially used by third parties.